src/analysis/vector/qgsgeometrysnappersinglesource.cpp

```diff
@@ -141,7 +142,7 @@ static bool snapPoint( QgsPoint *pt, QgsSpatialIndex &index, QVector<AnchorPoint
   QList<QgsFeatureId> fids = index.intersects( QgsRectangle( pt->x(), pt->y(), pt->x(), pt->y() ) );
   Q_ASSERT( fids.count() == 1 );
 
-  int spoint = fids[0];
+  int spoint = static_cast<int>( fids[0] );
```

Comment:
ouch, looks dangerous in general... but not caused by your changes!

Reply:
looking at buildSnapIndex(), there should be no harm at casting to int since this isn't really an arbitrary FID but the index to QVector<AnchorPoint>